scripts/generate_error_code.hs

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env stack
+-- stack --resolver lts-17.3 script --package string-interpolate --package universum
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+-- | Script for adding new error to ligo and also update the docs.
+module GenerateErrorCode where
+
+import Prelude ()
+import Universum
+import Data.String.Interpolate (i)
+
+data ErrorItem = ErrorItem
+  { eiLabel :: Text
+  , eiCode :: Integer
+  , eiDesc :: Text
+  }
+
+invalidInputErrors :: [ErrorItem]
+invalidInputErrors =
+  [ ErrorItem
+      { eiLabel = "invalid_witness_err"
+      , eiCode = 100
+      , eiDesc = "Invalid witness. The witness must refer to an initialized tick that is below or equal the supplied tick"
+      }
+  , ErrorItem
+      { eiLabel = "position_not_exist_err"
+      , eiCode = 101
+      , eiDesc = "Thrown by `collect_fees`: Position does not exist."
+      }
+  , ErrorItem
+      { eiLabel = "log_out_of_bounds_err"
+      , eiCode = 102
+      , eiDesc = "Thrown by `floor_log_half_bps`: Log out of bounds."
+      }
+  , ErrorItem
+      { eiLabel = "end_ladder_reached_err"
+      , eiCode = 103
+      , eiDesc = "Thrown by `half_bps_pow_rec`: Should not reach end of ladder."
+      }
+  , ErrorItem
+      { eiLabel = "past_deadline_err"
+      , eiCode = 104
+      , eiDesc = "Thrown by `x_to_y` and `y_to_x`: Past deadline."
+      }
+  , ErrorItem
+      { eiLabel = "smaller_than_min_asset_err"
+      , eiCode = 105
+      , eiDesc = "Threshold on amount of bought tokens violated: `dx` received < `min_dx` or `dy` received < `min_dy`"
+      }
+  , ErrorItem
+      { eiLabel = "asset_transfer_invalid_entrypoints_err"
+      , eiCode = 106
+      , eiDesc = "There is no appropriate transfer entrypoint in the provided token contract to transfer asset `x` or `y`."
```

Comment:
Yeah, should I create a new section for this kind of error?

Reply:
Hmm yeah, I think so. I think these sections make sense:

* Invalid Input Error Codes -> The user screwed up, they should do something about it
* Contract configuration errors -> the originator screwed up, they should do something about it
* Internal errors -> We screwed up  :see_no_evil:  we should do something about it

Reply:
@dcastro  I've added `Contract configuration errors` now, and move all internal errors code to `3**`. This should be ready for re-review.

Reply:
I think we should make this new "internal vs bad configuration" errors separation very-very clear because some of the internal errors are also possible due to the wrong configuration :thinking: 

> | 318 | `internal_sqrt_price_grow_err_1` | Thrown when `s.i_c < i_l.i` and the `sqrt_price` happened not to grow monotonically with tick indices (This is an invariant of the contract). |

This one can be violated if the initial storage already contains some ticks except for the edge ones.

> | 313 | `internal_position_underflow_err` | Number of positions underflow. |

This one can occur if initial storage has some positions (and `n_positions` field values do not respect them).

Reply:
Should I move these errors to configuration instead?